02.calendar/cal.rb

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+
+params = ARGV.getopts("y:m:")
+if !params["y"].nil? && !(1..9999).include?(params["y"].to_i)
+  puts "年は1~9999の中で指定してください" 
+  exit
+end
+if !params["m"].nil? && !(1..12).include?(params["m"].to_i)
+  puts "月は1~12の中で指定してください"
+  exit
+end
+
+if params["y"].nil?
+  year = Date.today.year
+else
+  year = params["y"].to_i
+end
+if params["m"].nil?
+  month = Date.today.month
+else
+  month = params["m"].to_i
+end
+
+first_date = Date.new(year, month, 1)
+last_date = Date.new(year, month, -1)
+
+calender_cells = Array.new(42)
+case first_date.cwday
+when 1
+  cell_of_first_day = 1
+when 2
+  cell_of_first_day = 2
+when 3
+  cell_of_first_day = 3
+when 4
+  cell_of_first_day = 4
+when 5
+  cell_of_first_day = 5
+when 6
+  cell_of_first_day = 6
+when 7
+  cell_of_first_day = 0
+end
+index = cell_of_first_day
+(1..last_date.day).each do |day|
+  calender_cells[index] = day
```

Comment:
プリミティブな整数値ではなく

```suggestion
(first_date..last_date).each do |date|
  calender_cells[index] = date
```

のようにDateオブジェクトを活用してみましょう

Reply:
date オブジェクトをそのまま使う形に修正しました！